lighthouse-core/lib/network-request.js

```diff
@@ -9,6 +9,31 @@
  * @fileoverview Fills most of the role of NetworkManager and NetworkRequest classes from DevTools.
  * @see https://cs.chromium.org/chromium/src/third_party/blink/renderer/devtools/front_end/sdk/NetworkRequest.js
  * @see https://cs.chromium.org/chromium/src/third_party/blink/renderer/devtools/front_end/sdk/NetworkManager.js
+
+  DevTools box-whisker
+
+    |-------[xxxxxXXXXXX]-|
+       (1)    (2)    (3) (4)
+
+  (1) Queuing (delta between renderer knowing about request and network manager knowing about it),
+      DNS, SSL, connection setup cost
+
+      CDP: left whisker edge is Network.requestWillBeSent timestamp
+
+  (2) light shaded region. browser network manager has initiated the request, hasn't recieved any bytes back yet
+      Note: even with early-hint response, only the "real" response is considered here
+
+      CDP: Network.responseRecieved timings.requestStart + timing.sendStart
+
+  (3) dark shaded region. browser network manager has recieved the very first header byte
+
+      CDP: Network.responseRecieved timings.requestStart + timing.recievedHeadersEnd
+      CDP: (right edge of box) Network.finished/Network.failed timestamp
+
+  (4) Trailing whisker: marks time when render process has used the resource.
+
+      Trace: ResourceFinished trace event finishedTime. Currently don't care about this. Could be
+      long if main thread is busy.
```

Comment:
```suggestion
  Details for network timing instrumentation in CDP and trace events: 
  https://nettimings.surge.sh/Network-Timings.svg (Aug 2022) 
  
  Also, see _updateTimingsForLightrider() for LR specifics.
```

Sorry to mess up your sweet ascii art, but i'd suggest we just invest in this one canonical document. sgty?

thanks for your help in working through all this. it's a real pita.

Reply:
surge links are liable to bit-rot so we shouldn't keep docs there. I'll pull this into the repo.

While there is tons of value in the thing you cooked up, I find myself overwhelmed when looking at it. I feel like a smaller view into the same info (As I have in the comment) is still useful to keep around for reference.

I'll need to double-check it all conforms with what's in your SVG, though.